src/Assert.php

```diff
@@ -1019,6 +1023,30 @@ public static function countBetween($array, $min, $max, $message = '')
         }
     }
 
+    public static function isList($array, $message = '')
+    {
+        if (!is_array($array) || !$array || array_keys($array) !== range(0, count($array) - 1)) {
+            static::reportInvalidArgument(
+                $message ?: 'Expected list - non-associative array.'
+            );
+        }
+    }
+
+    public static function isMap($array, $message = '')
+    {
+        if (
+            !is_array($array) ||
+            !$array ||
+            array_keys($array) !== array_filter(array_keys($array), function ($key) {
```

Comment:
There's no need for a lambda here, `array_filter(array_keys($array), 'is_string')` should suffice.